db/db_impl.cc

```diff
@@ -2487,6 +2487,12 @@ Status DBImpl::SetDBOptions(
             new_options.max_background_compactions, Env::Priority::LOW);
         MaybeScheduleFlushOrCompaction();
       }
+      if (new_options.delete_obsolete_files_period_micros !=
+          mutable_db_options_.delete_obsolete_files_period_micros) {
+        const uint64_t diff = new_options.delete_obsolete_files_period_micros -
```

Comment:
doing subtraction may get underflow (you are using unsigned int). How about have `delete_obsolete_files_last_run_` instead of `delete_obsolete_files_next_run_`.